### PR TITLE
fix:count error in route53

### PR DESCRIPTION
**Pull request by Aatishsharma77**
## what

- add for_each in main.tf (route53 resources)
-  update output.tf file

### CODE REVIEW

**Review by themaniskshah — APPROVED** [review]
LGTM!

File: main.tf [hunk 1]
```diff
@@ -85,22 +84,29 @@ data "aws_route53_zone" "default" {
...
-  ttl             = var.ttl
   allow_overwrite = var.allow_overwrite
-  name            = join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_name)
-  type            = join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_type)
-  records         = [join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_value)]
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = var.ttl
+  type            = each.value.type
+  zone_id         = join(",", data.aws_route53_zone.default[*].zone_id)
```
Comment (13archit) [inline main.tf:100]: Why we are using join function here? Also `[*]` can be replaced with `[0]` 'coz count for hosted zone data is set to 1.

File: main.tf [hunk 2]
```diff
@@ -85,22 +84,29 @@ data "aws_route53_zone" "default" {
...
 ##----------------------------------------------------------------------------------
 ## Most commonly, this resource is used together with aws_route53_record and aws_acm_certificate to request a DNS validated certificate, deploy the required validation records and wait for validation to complete.
 ##----------------------------------------------------------------------------------
 resource "aws_acm_certificate_validation" "default" {
-  count = var.enable && var.enable_dns_validation ? 1 : 0
-
-  certificate_arn         = join("", aws_acm_certificate.cert[*].arn)
-  validation_record_fqdns = aws_route53_record.default[*].fqdn
+  count                   = var.enable && var.enable_dns_validation ? 1 : 0
+  certificate_arn         = join(",", aws_acm_certificate.cert[*].arn)
```
Comment (13archit) [inline main.tf:108]: Why we are using join function here? Also [*] can be replaced with [0] 'coz count for certificate resource is set to 1.

Comment (Aatishsharma77) [reply main.tf:108]: because list of string is requird 
image

File: main.tf (hunk 1 above)
Comment (Aatishsharma77) [reply main.tf:100]: because list of string required
image

Comment (13archit) [reply main.tf:100]: if we directly use [*], i.e without join than also the result is list of string, so we can skip using join function.

File: main.tf (hunk 2 above)
Comment (13archit) [reply main.tf:108]: than we can skip using join function.

File: main.tf (hunk 1 above)
Comment (Aatishsharma77) [reply main.tf]: yes, you are right we can skip the join function.

File: main.tf (hunk 2 above)
Comment (Aatishsharma77) [reply main.tf]: yes, we can skip join function. Thank you..!!

**Review by 13archit — APPROVED** [review]

**Review by omsharma07 — APPROVED** [review]
LGTM

**Review by vaibhav7797 — APPROVED** [review]
LGTM!

**Review by themaniskshah — APPROVED** [review]
LGTM!

**Review by h1manshu98 — APPROVED** [review]
LGTM

**Pull request merged**